src/optimizer/input_column_deriver.cpp

```diff
@@ -82,18 +82,39 @@ void InputColumnDeriver::Visit(const QueryDerivedScan *op) {
           output_cols, {input_cols}};
 }
 
-void InputColumnDeriver::Visit(const PhysicalLimit *) { Passdown(); }
+void InputColumnDeriver::Visit(const PhysicalLimit *op) {
+  // Input columns and internal sort columns are needed by the child node
```

Comment:
I think you want say "aggregated columns" instead of "input columns"?

Reply:
I meant to say all aggregate expressions and TVEs in the required columns + internal sort columns.